src/components/AddLinkInput/AddLinkInput.tsx

```diff
@@ -0,0 +1,56 @@
+import { CSSProperties, useEffect, useRef, useState } from "react";
+import styles from "./AddLinkInput.module.scss";
+import InputIcon from "@/assets/svg/InputIcon";
+
+export default function AddLinkInput() {
+  const [isAddLinkInputVisible, setAddLinkInputVisible] = useState(true);
+  const addLinkInputRef = useRef(null);
+
+  const addLinkInputStyle: CSSProperties = !isAddLinkInputVisible
+    ? {
+        position: "fixed",
+        bottom: "0px",
+        padding: "24px 32px",
+        width: "100%",
+      }
+    : {};
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          setAddLinkInputVisible(entry.isIntersecting);
+        });
+      },
+      { threshold: 0.01 }
+    );
+
+    const addLinkInput = addLinkInputRef.current;
+    if (addLinkInput) {
+      observer.observe(addLinkInput);
+    }
```

Comment:
이런 경우에는 별도의 변수에 할당하기보다 직접 사용해주는게 더 보기 좋을 것 같네요~
```suggestion
    if (addLinkInputRef.current) {
      observer.observe(addLinkInputRef.current);
    }
```